### PR TITLE
[ENGA3-420]: Updated Omise_Capabilities class to prevent calling capabilities API when the keys are null.

### PR DESCRIPTION
**Pull request by aashishgurung**
#### 1. Objective

Prevent calling capabilities  API when the keys are null.

Jira Ticket: [#420](https://opn-ooo.atlassian.net/browse/ENGA3-420)

#### 2. Description of change

Updated `Omise_Capabilities` class to prevent calling capabilities API when the keys are null. Added a check that compares the newly fetched keys with the previously saved keys. If the keys are different then then we call to capabilities API else we return previously built instance.

**🔧 Environments:**

- WooCommerce: v6.8.0
- WordPress: v6.0.2
- PHP version: 8.1
- Omise plugin version: Omise-WooCommerce 4.23.3

### CODE REVIEW

**Review by new4762 — APPROVED** [review]
Thank you for your hard work. 🙇

File: includes/class-omise-capabilities.php [hunk 1]
```diff
@@ -8,30 +8,63 @@ class Omise_Capabilities {
...
+
+		// Do not call capabilities API if keys are not present
+		if($publicKey === null || $secretKey === null) {
+			return null;
+		}
+
+		if(self::$instance) {
+			$keysNotChanged = self::$instance->publicKey === $publicKey && self::$instance->secretKey === $secretKey;
+
+			// if keys are same then no need to call capabilities API as t
```
Comment (ajzkk) [inline includes/class-omise-capabilities.php:50]: did you forget to complete comment?

File: includes/class-omise-capabilities.php [hunk 2]
```diff
@@ -8,30 +8,63 @@ class Omise_Capabilities {
...
-				// suppressing error on the admin dashboard
-				return null;
+	public static function retrieve($pKey = null, $sKey = null)
+	{
+		$settings = Omise_Setting::instance();
+		$publicKey = !$pKey ? $settings->public_key() : $pKey;
+		$secretKey = !$sKey ? $settings->secret_key() : $sKey;
+
+		// Do not call capabilities API if keys are not present
+		if($publicKey === null || $secretKey === null) {
```
Comment (ajzkk) [inline includes/class-omise-capabilities.php:43]: Better to use `empty()` instead checking null?

**Review by ajzkk — APPROVED** [review]

File: includes/class-omise-capabilities.php (hunk 1 above)
Comment (aashishgurung) [reply includes/class-omise-capabilities.php]: @ajzkk No idea what happened here. I have add more context. Thanks.

File: includes/class-omise-capabilities.php (hunk 2 above)
Comment (aashishgurung) [reply includes/class-omise-capabilities.php]: @ajzkk Right. It better to check for any falsy value rather than just null. Thanks.

**Review by kan98 — APPROVED** [review]

**Review by FhanOmise — APPROVED** [review]

**Pull request merged**